core/chains/evm/logpoller/parser.go

```diff
@@ -268,15 +272,14 @@ func (v *pgDSLParser) whereClause(expressions []query.Expression, limiter query.
 			return "", errors.New("invalid cursor direction")
 		}
 
-		block, txHash, logIdx, err := valuesFromCursor(limiter.Limit.Cursor)
+		block, logIdx, _, err := valuesFromCursor(limiter.Limit.Cursor)
 		if err != nil {
 			return "", err
 		}
 
-		segment = fmt.Sprintf("%s AND block_number %s= :cursor_block AND tx_hash %s= :cursor_txhash AND log_index %s :cursor_log_index", segment, op, op, op)
+		segment = fmt.Sprintf("%s AND block_number %s= :cursor_block AND log_index %s :cursor_log_index", segment, op, op)
 
 		v.args.withField("cursor_block_number", block).
-			withField("cursor_txhash", common.HexToHash(txHash)).
```

Comment:
Isn't it misleading if we are ignoring txHash from the cursor? 
If we do this then there are 2 cases:
1. Reorg happened which changed the txHash but log poller handled it somehow and logs are returned(which shouldn't ever happen since we are limiting this to finalized logs)
2. No reorg, all good

Similar cases if we also filter logs by the txHash
1. Reorg happened, logPoller handled it, but we return an error here since txHash doesn't match
2. Reorg happened, logPoller didn't handle it and the caller gets an error ( or log poller is in degraded state which is basically the same for the caller)
3. No reorg, all good

Is there still a discussion to be had here on how to handle this? I am not a fan of passing in txHash and then not using it. That said this might also be fine, but I would leave some comments around this code to remind us of this

Reply:
I thought the idea was that we didn't need to be concerned with reorgs because we aren't allowing cursors to be used with anything other than finalized.

Reply:
Yes, but why not also give extra info?

Reply:
The `txHash` is already in the cursor, so the amount of info doesn't change, per our conversation. Adding the `txHash` to the query behind the scenes becomes an irrelevant no-op when limiting to finalized blocks and won't affect precision.

Reply:
> Yes, but why not also give extra info and be more precise?

I think I may have misunderstood what @ilija42 was suggesting in our last meeting. I am okay with returning tx hash as a result, as attaching extra irrelevant info can't hurt anything.  But I'm not okay with using it to restrict the queries, as that would lead to unpredictable wrong results... making it easy to miss logs.

Reply:
> Isn't it misleading if we are ignoring txHash from the cursor?

I agree that it's counter-intuitive... ideally, we should not be passing it to the cursor as in the best case scenario it would not affect anything and in the worse case it would do actual harm... in no case would it be able to improve the results as far as I can see.